### ISSUE TITLE
Issue with explicit_odes then update_source

### ISSUE THREAD

**Issue by SCarter79** [pharmpy/pharmpy#79]
If i run the following code (run3.mod attached), i get an issue when followed by update_source. This was run on a jupyter notebook running python 3.8.5 and pharmpy 0.12.0.

```
# Import packages
from pharmpy.modeling import update_source
from pharmpy.modeling import explicit_odes

# copy model3 so that run3.mod is not overwritten
model3 = Model('run3.mod')
model31 = model3.copy()

# convert to an ODE system and update the nonmem file
explicit_odes(model31)
update_source(model31)
```

gives the following error - which looking at the bottom seems to be a transit compartment issue (which was generated in pharmpy to create model 3):

```
---------------------------------------------------------------------------
UnexpectedCharacters                      Traceback (most recent call last)
<ipython-input-4-9e3a148f7d76> in <module>
     12 
     13 explicit_odes(model31)
---> 14 update_source(model31)
     15 
     16 # parset_31 = model31.parameters

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\modeling\common.py in update_source(model)
     24     changes in the model object.
     25     """
---> 26     model.update_source()
     27     return model
     28 

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\plugins\nonmem\model.py in update_source(self, path, force, nofiles)
    107             self.statements  # Read statements unless read
    108         if hasattr(self, '_statements'):
--> 109             update_statements(self, self._old_statements, self._statements, trans)
    110             self._old_statements = self._statements.copy()
    111 

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\plugins\nonmem\update.py in update_statements(model, old, new, trans)
    363         old_odes = old.ode_system
    364         if new_odes != old_odes:
--> 365             update_ode_system(model, old_odes, new_odes)
    366 
    367     after_odes = False

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\plugins\nonmem\update.py in update_ode_system(model, old, new)
    280     """
    281     if type(old) == CompartmentalSystem and type(new) == ExplicitODESystem:
--> 282         to_des(model, new)
    283     elif type(old) == CompartmentalSystem and type(new) == CompartmentalSystem:
    284         # subs = model.control_stream.get_records('SUBROUTINES')[0]

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\plugins\nonmem\update.py in to_des(model, new)
    343     subs.append_option('TOL', 3)
    344     des = model.control_stream.insert_record('$DES\nDUMMY=0')
--> 345     des.from_odes(new)
    346     mod = model.control_stream.insert_record('$MODEL\n')
    347     for eq, ic in zip(new.odes[:-1], list(new.ics.keys())[:-1]):

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\plugins\nonmem\records\code_record.py in from_odes(self, ode_system)
    446             expression = ode.rhs.subs(function_map)
    447             statements.append(Assignment(symbol, expression))
--> 448         self.statements = statements
    449 
    450     def __str__(self):

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\plugins\nonmem\records\code_record.py in statements(self, new)
    261                 else:
    262                     statement_str = f'\n{repr(s).replace(":", "")}'
--> 263                 node_tree = CodeRecordParser(statement_str).root
    264                 node = node_tree.all('statement')[0]
    265                 if node_index == 0:

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\parse_utils\generic.py in __init__(self, buf, **lark_options)
    468     def __init__(self, buf=None, **lark_options):
    469         self.lark_options.update(lark_options)
--> 470         self.root = self.parse(buf)
    471 
    472     def parse(self, buf):

~\Anaconda3\envs\pharmpy\lib\site-packages\pharmpy\parse_utils\generic.py in parse(self, buf)
    480             return None
    481 
--> 482         root = self.lark.parse(self.buffer)
    483         if self.non_empty:
    484             root = self.insert(root, self.non_empty)

~\Anaconda3\envs\pharmpy\lib\site-packages\lark\lark.py in parse(self, text, start, on_error)
    493 
    494         try:
--> 495             return self.parser.parse(text, start=start)
    496         except UnexpectedInput as e:
    497             if on_error is None:

~\Anaconda3\envs\pharmpy\lib\site-packages\lark\parser_frontends.py in parse(self, text, start)
    221 
    222     def parse(self, text, start):
--> 223         return self._parse(start, text)
    224 
    225 class XEarley_CompleteLex(XEarley):

~\Anaconda3\envs\pharmpy\lib\site-packages\lark\parser_frontends.py in _parse(self, start, input, *args)
     70                 raise ValueError("Lark initialized with more than 1 possible start rule. Must specify which start rule to parse", start)
     71             start ,= start
---> 72         return self.parser.parse(input, start, *args)
     73 
     74 

~\Anaconda3\envs\pharmpy\lib\site-packages\lark\parsers\earley.py in parse(self, stream, start)
    292                 columns[0].add(item)
    293 
--> 294         to_scan = self._parse(stream, columns, to_scan, start_symbol)
    295 
    296         # If the parse was successful, the start

~\Anaconda3\envs\pharmpy\lib\site-packages\lark\parsers\xearley.py in _parse(self, stream, columns, to_scan, start_symbol)
    137             self.predict_and_complete(i, to_scan, columns, transitives)
    138 
--> 139             to_scan = scan(i, to_scan)
    140 
    141             if token == '\n':

~\Anaconda3\envs\pharmpy\lib\site-packages\lark\parsers\xearley.py in scan(i, to_scan)
    114 
    115             if not next_set and not delayed_matches and not next_to_scan:
--> 116                 raise UnexpectedCharacters(stream, i, text_line, text_column, {item.expect.name for item in to_scan}, set(to_scan))
    117 
    118             return next_to_scan

UnexpectedCharacters: No terminal defined for '(' at line 2 col 37

DADT(1) = -A(1)*K23 + K12*A_TRANSIT1(t)
                                    ^

Expecting: {'CALL', 'NAME', 'STAR', 'VERBATIM_STATEMENT', 'RETURN', 'DADT', 'SLASH', 'A', 'MINUS', 'WS', 'EXIT', '__ANON_17', 'COMMENT', '__ANON_16', 'IF', 'PLUS', '__ANON_18', 'LF'}

```

NONMEM code for run3.mod:

```
$PROBLEM    MOXONIDINE 1 COMP PK MODEL 

$INPUT   ID VISI XAT2=DROP DGRP=DROP DOSE=DROP FLAG=DROP ONO=DROP
       XIME=DROP DVO=DROP NEUY=DROP SCR=DROP AGE SEX NYHA=DROP WT
      DROP ACE DIG DIU NUMB=DROP XTAD TIME VIDD=DROP CLCR AMT SS
      II DROP CMT=DROP CONO=DROP DV EVID OVID=DROP SHR=DROP
            SHR2=DROP
$DATA      mx19B.csv IGNORE=@ 
$SUBROUTINE ADVAN5 TRANS1
$MODEL COMPARTMENT=(TRANSIT1 DEFDOSE) COMPARTMENT=(DEPOT) COMPARTMENT=(CENTRAL)
$PK
;---------- PK MODEL --------------

MDT = THETA(4)
CL    = THETA(1)*EXP(ETA(1))  ; TYPICAL CL VALUE
V     = THETA(2)*EXP(ETA(2))  ; TYPICAL V VALUE
MAT   = THETA(3)*EXP(ETA(3))  ; TYPICAL MAT VALUE
K30 = CL/V
KA    = 1/MAT
VC    = V
K12 = 1/MDT
K23 = KA

$ERROR (ONLY OBSERVATIONS)
IPRED = A(3)/VC
CONC = IPRED

Y = CONC + CONC*EPS(1) + EPS(2)

$THETA  (0, 26.3)           ; POP_CL
$THETA  (0, 1.4)            ; POP_V
$THETA  (0, 4.2)            ; POP_MAT

$THETA  (0,0.125) ; POP_MDT
$OMEGA   0.1                ; IIV_CL
$OMEGA   0.1                ; IIV_V
$OMEGA   0.1                ; IIV_MAT

$SIGMA  0.1                 ; PROP
$SIGMA  0.1                 ; ADD

$ESTIMATION METHOD=1 INTER MAXEVALS=9999
$COVARIANCE PRINT=E
$TABLE      ID TIME VISI ONEHEADER NOPRINT FILE=sdtab3
$TABLE      ID CL V MAT ETA(1) ETA(2) ETA(3) ONEHEADER NOPRINT
            FILE=patab3

```

**Comment by rikardn** [comment]
Thanks for reporting this. It is indeed a bug.

**Comment by SCarter79** [comment]
this seems to be only if there is a transit compartment as it worked fine for a two compartment model without a transit compartment

**Comment by rikardn** [comment]
I couldn't reproduce the issue with your example. It might have been fixed as a side effect of some other fix. However I found that $MODEL was duplicated so I fixed that instead (with bc5f0a07400ee9ba51f85eb93976615ddf61b8f3). Please reopen or report again if the problem remains in the next version.
